addons/project_todo/static/src/views/todo_list/todo_list_renderer.js

```diff
@@ -0,0 +1,20 @@
+import { ListRenderer } from '@web/views/list/list_renderer';
+
+import { useEffect } from "@odoo/owl";
+
+export class TodoListRenderer extends ListRenderer {
+    setup() {
+        super.setup();
+        useEffect(
+            () => this.focusName(this.props.list.editedRecord),
+            () => [this.props.list.editedRecord]
+        );
+    }
+
+    focusName(editedRecord) {
+        if (editedRecord?.isNew && !editedRecord.dirty) {
+            const col = this.columns.find((c) => c.name === "name");
+            this.focusCell(col);
+        }
+    }
+}
```

Comment:
To avoid code duplication, we need to modify the SubtaskListRenderer and utilize its renderer here.thanks